### ISSUE TITLE
Resolve images source while generating ISD

### ISSUE THREAD

**Issue by bbert** [sandflow/imscJS#241]
Question: is there any reason for resolving images source while rendering HTML and not while generating ISD?

Resolving images at ISD generation step would ease the comparison of image subtitles before rendering the cues, by comparing the ISD contents.
And comparing subtitles may be useful to handle contigous subtitles with same content (same image).

**Comment by palemieux** [comment]
The ISD includes the `smpte:backgroundImage` attribute values and the application controls the image resolver, so it should be possible to compare images before HTML generation.

**Comment by bbert** [comment]
That means that you have to go through the returned ISD tree and resolve the sources.
Thus, these sources would be resolved twice: once while parsing the returned ISD, and once while rendering from within imscJS library.
I was just wondering if it could be more convenient to resolve while generating ISD, except if there was any valid reason I am not aware of.

**Comment by palemieux** [comment]
The assumption is that one can compare the value `smpte:backgroundImage` to determine whether images are identical, without resolving the image to a specific location. It might help if you shared a specific use case you have in mind.

**Comment by bbert** [comment]
Yes but the `smpte:backgroundImage` refers to an image id which can be different accross the chunks/segments but containing the same image data.
One specific use case is chunked live low latency streams, with consecutive chunks containing same image subtitle but with different id.

**Comment by bbert** [comment]
No, agree this is not awesome, but for live low latency and chunked transfer encoding, you may have no other solution.
Thus you could compare the resolved URL or the base64 PNG representation

**Comment by bbert** [comment]
Sorry, I believed I already described this.
On browsers, contiguous cues with same content will produce some blinking effect, due to same content being hidden then immediatly rendered.
In dash.js we already implemented a patch to avoid this (https://github.com/Dash-Industry-Forum/dash.js/pull/4103), but that works only for text subtitles.

**Comment by palemieux** [comment]
Does the blinking effect persist when the `src` attribute of the `img` element is modified, instead of a new `img` element created?

**Comment by nigelmegitt** [comment]
I'd suggest a mechanism for doing this would be to construct a data url from the dereferenced image resource, since the text of the data url can be compared directly, and that data url can be passed into the HTML also, avoiding a second fetch.
